lib/timers/events.rb

```diff
@@ -60,18 +60,12 @@ def schedule(time, callback)
       # Maintain sorted order, O(logN) insertion time.
       @sequence.insert(index, handle)
       
-      return handle
+      handle
     end
     
     # Returns the first non-cancelled handle.
     def first
-      while handle = @sequence.last
-        if handle.cancelled?
-          @sequence.pop
-        else
-          return handle
-        end
-      end
+      @sequence.reverse.find { |handle| !handle.cancelled? }
```

Comment:
@tarcieri added a volume testing benchmark with 20 concurrent timers firing 300 times each:

With change:
```
Serviced 3766885 events in 100.282 seconds, 37562.922558385355 e/s; across 20 timers.
Serviced 3766843 events in 100.192 seconds, 37596.24520919834 e/s; across 20 timers.
```

Without change:
```
Serviced 3766828 events in 100.093 seconds, 37633.281048624776 e/s; across 20 timers.
Serviced 3766936 events in 100.191 seconds, 37597.54868201735 e/s; across 20 timers.
```

The number of tests run is fluxating for some reason, and this machine I'm running it on is highly variable in load, so the tests are there but the results are inconclusive. I will push the benchmark.